vignettes/IntroToAnnotationPackages.Rmd

```diff
@@ -0,0 +1,366 @@
+(e.g.`r Biocpkg("org.Sc.sgd.db")`where \<Ab>\ is a 2-letter abbreviation of the
+organism (e.g.`r Biocpkg("Sc")` for *Saccharomyces cerevisiae*) and \<id>\ is an
+abbreviation (in lower-case) describing the type of central identifier (e.g.
+`r Biocpkg("sgd")` for gene identifiers assigned by the Saccharomyces Genome
+Database, or `r Biocpkg("eg")` for Entrez Gene ids).
+
+Just as the chip packages load a `ChipDb` object, the org packages will load a
+`OrgDb` object. The following exercise should acquaint you with the use of these
+methods in the context of an organism package.
+
+**Exercise 1**
+
+Display the `OrgDb` object for the `r Biocpkg("org.Hs.eg.db")` package.
+
+Use the `columns` method to discover which sorts of annotations can be extracted
+from it. Is this the same as the result from the `keytypes` method?  Use the
+`keytypes` method to find out.
+
+Finally, use the `keys` method to extract UNIPROT identifiers and then pass
+those keys in to the `select` method in such a way that you extract the gene
+symbol and KEGG pathway information for each.  Use the help system as needed to
+learn which values to pass in to columns in order to achieve this.
+
+**Solution:**
+
+```{r}
+library(org.Hs.eg.db)
+columns(org.Hs.eg.db)
+```
+
+```{r}
+help("SYMBOL") ## for explanation of these columns and keytypes values
+keytypes(org.Hs.eg.db)
+```
+
+```{r}
+uniKeys <- head(keys(org.Hs.eg.db, keytype="UNIPROT"))
+cols <- c("SYMBOL", "PATH")
+select(org.Hs.eg.db, keys=uniKeys, columns=cols, keytype="UNIPROT")
+```
+
+So how could you use select to annotate your results? This next exercise should
+hlep you to understand how that should generally work.
+
+**Exercise 2**
+
+Please run the following code snippet (which will load a fake data result that I
+have provided for the purposes of illustration):
+
+```{r selectData}
+load(system.file("extdata", "resultTable.Rda", package="AnnotationDbi"))
+head(resultTable)
+```
+
+The rownames of this table happen to provide entrez gene identifiers for each
+row (for human). Find the gene symbol and gene name for each of the rows in
+resultTable and then use the merge method to attach those annotations to it.
+
+
+**Solution:**
+```{r}
+annots <- select(org.Hs.eg.db, keys=rownames(resultTable),
+columns=c("SYMBOL","GENENAME"), keytype="ENTREZID")
+
+resultTable <- merge(resultTable, annots, by.x=0, by.y="ENTREZID")
+head(resultTable)
+```
+
+# Using select with GO.db
+
+When you load the GO.db package, a `GODb` object is also loaded. This allows you
+to use the `columns`, `keys`, `keytypes` and `select` methods on the contents of
+the GO ontology. So if for example, you had a few GO IDs and wanted to know more
+about it, you could do it like this:
+
+```{r selectGO}
+library(GO.db)
+GOIDs <- c("GO:0042254","GO:0044183")
+select(GO.db, keys=GOIDs, columns="DEFINITION", keytype="GOID")
+```
+
+# Using select with TxDb packages
+
+A `TxDb` package (a 'TxDb' package) connects a set of genomic coordinates to
```

Comment:
Sorry @BerylKanali Please change it to 

```
A `TxDb` package (a 'TxDb' package) connects a set of genomic coordinates to
```

since this link is not working